messaging/src/main/java/org/axonframework/tracing/TracingHandlerEnhancerDefinition.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing;
+
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.HandlerEnhancerDefinition;
+import org.axonframework.messaging.annotation.MessageHandlingMember;
+import org.axonframework.messaging.annotation.WrappedMessageHandlingMember;
+
+import java.lang.reflect.Executable;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+/**
+ * Enhances message handlers with the provided {@link SpanFactory}, wrapping handling of the message in a {@link Span}
+ * that is reported to the monitoring tooling.
+ * <p>
+ * Since {@code EventSourcingHandlers} can be very noisy when loading the aggregate, they can be enabled or disabled
+ * separately through constructor configuration.
+ *
+ * @since 4.6.0
+ * @author Mitchell Herrijgers
+ */
+public class TracingHandlerEnhancerDefinition implements HandlerEnhancerDefinition {
```

Comment:
But, why wasn't it added to the `org.axonframework.messaging.annotation.HandlerEnhancerDefinition` file in the `resources/META-INF` folder?
Right now, the only way it's configured for users is through Spring Boot.

Reply:
Unfortunately, that wouldn't be possible. It would create a circular dependency between the HandlerEnhancerDefinition and the Configuration. Also, the ServiceLoader uses the no-arg constructor to create instances.

Reply:
I think this is a reasonable compromise for now...

Reply:
We could create an interface that can be used by the ServiceLoader as well:

```java
@FunctionalInterface
public interface HandlerEnhancerDefinitionFactory {

    HandlerEnhancerDefinition create(Configuration configuration);
}
```

This could lead to people defining something like this, making it usable in CDI environments:
```java
public class TracingHandlerEnhancerDefinitionFactory implements HandlerEnhancerDefinitionFactory {

    @Override
    public HandlerEnhancerDefinition create(Configuration configuration) {
        return new TracingHandlerEnhancerDefinition(
                OpenTelemetrySpanFactory.builder().build(), false
        );
    }
}
```

The way they get or create the `SpanFactory` is up to the user. Since they are lightweight and hold no state, there is also no problem in two separate span factories; one in the configuration, and another in the handler enhancer definition. 
I understand this is a lot to implement for a small use case, but we should consider this in the future, especially if we want to be more container-agnostic

Reply:
You can already define `ConfigurerModule` using the service loader. These are ignored when using Spring, as it initialized the Configurer explicitly to ignore those. In spring, instead, the Application Context is used to load these modules.
There would have to be a way to register these handler enhancers on the Configurer, though...

Reply:
We could change the enhancer definitions to be on the Configurer in Axon 5.

In the meantime, it could also be possible to configure them on the `Configurer`, by creating a `MultiHandlerDefinition` yourself. I imagine that's not really user-friendly though

Reply:
Hmm, all kinds of bothersome. Let's leave this as a known misser for now. Be sure to document this in the Reference Guide though!